libs/dirty-check-forms/src/lib/dirty-check.ts

```diff
@@ -59,16 +60,18 @@ export function dirtyCheck<U>(
       shareReplay({ bufferSize: 1, refCount: true })
     );
 
-    observer.add(
-      fromEvent(window, 'beforeunload')
-        .pipe(withLatestFrom(isDirty$))
-        .subscribe(([event, isDirty]) => {
-          if (isDirty) {
-            event.preventDefault();
-            event.returnValue = false;
-          }
-        })
-    );
+    if (window) {
```

Comment:
Why not using isPlatformBrowser, and that's it?

Reply:
I think it's better not to use globals and pass all module dependencies as parameters. Also, dirty check doesn't support checking without using the `beforeunload` event. Maybe `useBeforeunloadEvent` parameter can be used as a compromise?

Reply:
Yes, we can add useBeforeunloadEvent, but use isPlatformBrowser to protect from SSR